### ISSUE TITLE
Ambigious InitializeComponent()?

### ISSUE THREAD

**Issue by Zerowalker** [dotnet/vscode-csharp#3254]
## Issue Description ##
MainWindow.xml.cs keeps complaining about "InitializeComponent();"

"The call is ambiguous between the following methods or properties: 'MainWindow.InitializeComponent()' and 'MainWindow.InitializeComponent()' (CS0121) [ag_watch]"

It seem to look at "obj/MainWindow.g.cs" and "obj/MainWindow.g.i.cs",
which might be why it seems amiguity, but not sure.

## Steps to Reproduce ##
Hard to tell, it just happens, but it's fairly quick.

## Expected Behavior ##
Not complaining about it as there shouldn't be any issues as far as i can tell.

## Actual Behavior ##
It complaining about ambiguity while there isn't any

### OmniSharp log ###
<details>Starting OmniSharp server at 8/30/2019, 1:30:17 AM
    Target: ....\ag_watch

OmniSharp server started.
    Path: C:\Users\ZeCurator\.vscode-insiders\extensions\ms-vscode.csharp-1.21.1\.omnisharp\1.34.3-beta.22\OmniSharp.exe
    PID: 19944

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on Windows 6.2.9200.0 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 2 MSBuild instance(s)
            1: Visual Studio Build Tools 2017 15.9.28307.812 - "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin"
            2: StandAlone 16.0 - "C:\Users\ZeCurator\.vscode-insiders\extensions\ms-vscode.csharp-1.21.1\.omnisharp\1.34.3-beta.22\.msbuild\Current\Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to 'C:\Users\ZeCurator\.vscode-insiders\extensions\ms-vscode.csharp-1.21.1\.omnisharp\1.34.3-beta.22\.msbuild\Current\Bin\MSBuild.exe'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 16.0 - "C:\Users\ZeCurator\.vscode-insiders\extensions\ms-vscode.csharp-1.21.1\.omnisharp\1.34.3-beta.22\.msbuild\Current\Bin"
            CscToolExe = csc.exe
            CscToolPath = C:\Users\ZeCurator\.vscode-insiders\extensions\ms-vscode.csharp-1.21.1\.omnisharp\1.34.3-beta.22\.msbuild\Current\Bin\Roslyn
            MSBuildExtensionsPath = C:\Users\ZeCurator\.vscode-insiders\extensions\ms-vscode.csharp-1.21.1\.omnisharp\1.34.3-beta.22\.msbuild
            MSBuildToolsPath = C:\Users\ZeCurator\.vscode-insiders\extensions\ms-vscode.csharp-1.21.1\.omnisharp\1.34.3-beta.22\.msbuild\Current\Bin
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '....\ag_watch'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '....\ag_watch'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '....\ag_watch\ag_watch.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '....\ag_watch'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: ....\ag_watch\ag_watch.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.EditorConfigWorkspaceOptionsProvider, Order: 200
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '....\ag_watch' on host 19712.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '....\ag_watch\ag_watch.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '....\ag_watch\ag_watch.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: ag_watch
[info]: OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.CachingCodeFixProviderForProjects
        Skipping code fix provider 'Microsoft.CodeAnalysis.Analyzers.MetaAnalyzers.Fixers.ApplyDiagnosticAnalyzerAttributeFix' because it's language doesn't match 'C#'.
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '....\ag_watch\ag_watch.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '....\ag_watch\ag_watch.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: ....\ag_watch\ag_watch.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '....\ag_watch\ag_watch.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: ag_watch
[info]: OmniSharp.Roslyn.CSharp.Services.Refactoring.V2.RunCodeActionService
        Applying code action: using Microsoft.Extensions.Configuration;
</details>

### C# log ###
<details>Getting latest OmniSharp version information
Downloading package 'Latest OmniSharp Version Information' (1 KB).................... Done!
</details>

## Environment information ##

**VSCode version**: 1.38.0-insider
**C# Extension**: 1.21.1

<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview8-013656
 Commit:    8bf06ffc8d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview8-013656\

Host (useful for support):
  Version: 3.0.0-preview8-28405-07
  Commit:  d01b2fb7bc

.NET Core SDKs installed:
  3.0.100-preview8-013656 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.0-preview8.19405.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.0-preview8-28405-07 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0-preview8-28405-07 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|csharp|ms-vscode|1.21.1|
|Go|ms-vscode|0.11.4|
|powershell-preview|ms-vscode|2019.5.0|
|remote-containers|ms-vscode-remote|0.71.0|
|remote-ssh|ms-vscode-remote|0.45.6|
|remote-ssh-edit|ms-vscode-remote|0.45.6|
|remote-ssh-explorer|ms-vscode-remote|0.45.6|
|remote-wsl|ms-vscode-remote|0.39.4|
|shell-format|foxundermoon|6.1.1|
|vetur|octref|0.22.2|
|vscode-remote-extensionpack|ms-vscode-remote|0.16.0|
|vscode-xml-complete|rogalmic|0.1.9|;

</details>

**Comment by mycelo1** [comment]
That happens to me whenever I run a WPF application. Deleting the g.i.cs files inside obj folder seem to solve the problem temporarily. Ominisharp should ignore these files.

**Comment by Zerowalker** [comment]
Deleting those doesn't do anything for me it seems.
But it does seem to be related to the files inside obj, as it seems to look at them and hence see duplicates.
Kinda odd it just complains on the InitialComponent() though.

Would love to have this solved as it's quite annoying to always have the error and you can't tell if the "red file" is an actual error or not until you open and notice it's just that InitialComponent one;P

I would gladly debug etc if that helps, i am currently running the latest beta as i think someone mentioned it being solved there, but that sadly wasn't the case.

So well if i can do anything to help out do tell,
cause i don't want to just demand a solution without providing some kind of assistance myself:)

(Great software overall btw, i don't mean to sound negative about it!)

**Comment by vijay7790** [comment]
Today I got the same issue. It seems omnisharp is not able to find settings to exclude obj and bin folders. I created omnisharp,json file in the %USERPROFILE%/.omnisharp/omnisharp.json directory,

and included below mentioned settings in omnisharp.json
{
  "fileOptions": {
    "systemExcludeSearchPatterns": [
      "**/bin/**/*",
      "**/obj/**/*"
    ],
    "userExcludeSearchPatterns": []
  }
}


Then after restarting vscode, the issue got resolved.

**Comment by cartermp** [comment]
Trying a new wpf app with .NET Core 3.1 + latest plugin builds + runs with no IntelliSense issues. Can you reproduce this in that environment?

**Comment by SirIntruder** [comment]
I believe this has something to do with this: https://github.com/OmniSharp/omnisharp-roslyn/pull/1704

**Comment by GitClickOk** [comment]
The workaround from @vijay7790 worked for me.

To reproduce the bug, I have created a basic WPF app with:
`dotnet new wpf`
No problems until here. So I add a package like Dapper:
`dotnet add package Dapper`
and now this bug happens.

**Comment by karimrahimian** [comment]
sdff

**Comment by AskYous** [comment]
@karimrahimian You may be onto something there.

**Comment by JoeRobich** [comment]
This could be an O# bug. If you are still seeing this problem, please open a new issue at https://github.com/OmniSharp/omnisharp-roslyn/issues
